File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
    var totalBirds int
	for i := 0; i < len(birdsPerDay); i++ {
        totalBirds += birdsPerDay[i]
    }
    return totalBirds
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
    var birdsCounted int
    var counts int
	for i := week * 7 - 7; counts < 7; i++ {
        birdsCounted += birdsPerDay[i]
        counts += 1
    }
    return birdsCounted
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
    for i := 0; i < len(birdsPerDay); i += 2 {
        birdsPerDay[i] += 1
    }
    return birdsPerDay
}

File: solutions/javascript/bird-watcher/2/bird-watcher.js
// @ts-check
//
// The line above enables type checking for this file. Various IDEs interpret
// the @ts-check directive. It will give you helpful autocompletion when
// implementing this exercise.

/**
 * Calculates the total bird count.
 *
 * @param {number[]} birdsPerDay
 * @returns {number} total bird count
 */
export function totalBirdCount(birdsPerDay) {
    let sum = 0
    for (let i=0; i < birdsPerDay.length; i++) {
	sum += birdsPerDay[i]
    }
    return sum
}

/**
 * Calculates the total number of birds seen in a specific week.
 *
 * @param {number[]} birdsPerDay
 * @param {number} week
 * @returns {number} birds counted in the given week
 */
export function birdsInWeek(birdsPerDay, week) {
    let birdsCounted = 0;
    let counts = 0;
    for (let i = week * 7 - 7; counts < 7; i++) {
	birdsCounted += birdsPerDay[i]
	counts += 1
    }
    return birdsCounted
}

/**
 * Fixes the counting mistake by increasing the bird count
 * by one for every second day.
 *
 * @param {number[]} birdsPerDay
 * @returns {void} should not return anything
 */
export function fixBirdCountLog(birdsPerDay) {
    for (let i = 0; i < birdsPerDay.length; i += 2) {
	birdsPerDay[i] += 1
    }
}
	
